### DISCUSSION TITLE
pdm for libraries

### DISCUSSION THREAD

**Discussion by tkukushkin** [pdm-project/pdm#1362] [Q&A]
Suppose I make a library for Django and want to test it on different Django versions using tox/nox. How can I do this when using pdm? Why do I need a pdm.lock file at all when developing a library?

**Comment by pawamoy** [comment]
Usage with Nox is documented here: https://pdm.fming.dev/latest/usage/advanced/#use-nox-as-the-runner. You can probably override the Django version in each run with a pip install through Nox.

The lock file is not strictly necessary. You can see it as a cache file storing the resolved dependencies. It's useful to avoid re-resolving deps each time you want to install them.
